### ISSUE TITLE
using TapNet without labels in input

### ISSUE THREAD

**Issue by homriidan** [xuczhang/tapnet#7]
Hi,
I tried the TapNet with my own dataset scenario and got very good results on training and very poor on validation.

When I moved to validation (model.eval), I noted that the model is getting the labels as part of input for both training and validation phases, in order to calculate the attention and prototypical: 
https://github.com/xuczhang/tapnet/blob/master/models.py#L179
https://github.com/xuczhang/tapnet/blob/master/train.py#L217

I wrapped the part of code calculates x_proto (under self.training condition)
and saved the x_proto for validation phase using pytorch register_buffer. 
https://github.com/xuczhang/tapnet/blob/master/models.py#L176:L212

The problem is that i get very poor results in validation (high loss), far away from training, 
which of-course not happening if i leave the validation phase with labels.

How shall I use this model for prediction without passing labels? (as you used in test() func inputs)
Can you adapt the test() / new predict function to get input without labels after model training with saved prototypicals?

Thanks!

**Comment by helloMickey** [comment]
Have you solved your problem? I also tried TapNet with my custom dataset, and I find **some serious problems** in the source code (see my description -> [issue](https://github.com/xuczhang/tapnet/issues/9) ).
